### PR TITLE
feat(Discord.style): update presence

### PR DESCRIPTION
**Pull request by DiscordAnaxes**
## Description 

Yearly update of some of my presences.

Version 1.0.12, updates and/or implements the following (Extremely small update): 
- Updated static page system (now using a record)

I apologise if something is incorrect. I haven't used PreMiD in a long time. 

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary>Small updates</summary>

Image_1
Image_2

</details>

### CODE REVIEW

File: websites/D/Discord.style/presence.ts [hunk 1]
```diff
@@ -4,12 +4,28 @@ const presence = new Presence({
 	browsingTimestamp = Math.floor(Date.now() / 1000);
 
 presence.on("UpdateData", async () => {
+	let presenceData: PresenceData = {
+		largeImageKey: "discordstyle_logo",
```
Comment (theusaf) [inline websites/D/Discord.style/presence.ts:8]: Imgur links are preferred over Discord asset names.
```suggestion
		largeImageKey: "https://i.imgur.com/1wKlRwf.png",
```

File: websites/D/Discord.style/presence.ts [hunk 2]
```diff
@@ -4,12 +4,28 @@ const presence = new Presence({
...
-		presenceData: PresenceData = {
-			largeImageKey: "discordstyle_logo",
+		pages: Record<string, PresenceData> = {
+			"/bot": {
+				details: "Viewing 🤖 bot page",
+				buttons: [{ label: "View Page", url: document.location.href }],
+			},
+			"/search": {
+				details: "Viewing search page",
+				buttons: [{ label: "View Page", url: document.location.href }],
```
Comment (theusaf) [inline websites/D/Discord.style/presence.ts:20]: `document.location` properties are accessed many times. For consistency, they should be destructured.

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/D/Discord.style/presence.ts [hunk 3]
```diff
@@ -4,23 +4,37 @@ const presence = new Presence({
...
-		presenceData: PresenceData = {
-			largeImageKey: "discordstyle_logo",
+		pages: Record<string, PresenceData> = {
+			"/bot": {
+				details: "Viewing 🤖 bot page",
+				buttons: [{ label: "View Page", url: href }],
+			},
+			"/search": {
+				details: "Viewing 🔎 search page",
+				buttons: [{ label: "View Page", url: href }],
```
Comment (DarkVillager) [inline websites/D/Discord.style/presence.ts:21]: no don't do that

File: websites/D/Discord.style/presence.ts [hunk 4]
```diff
@@ -4,23 +4,37 @@ const presence = new Presence({
...
 
 presence.on("UpdateData", async () => {
-	const showTimestamp = await presence.getSetting<boolean>("timestamp"),
+	let presenceData: PresenceData = {
+		largeImageKey: "https://i.imgur.com/1wKlRwf.png",
+	};
+
+	const { href, pathname } = document.location,
+		showTimestamp = await presence.getSetting<boolean>("timestamp"),
 		showButtons = await presence.getSetting<boolean>("buttons"),
```
Comment (DarkVillager) [inline websites/D/Discord.style/presence.ts:13]: Make this one promise.

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/D/Discord.style/presence.ts (hunk 3 above)
Comment (EncryptedDev) [reply websites/D/Discord.style/presence.ts:23]: Agreed, arbitrary buttons are not allowed

Comment (DiscordAnaxes) [reply websites/D/Discord.style/presence.ts:23]: Alright, will change.

**Review by EncryptedDev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**